api/src/public/login/password-reset.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable */
+import {
+  setState,
+  request,
+  getLocale,
+  parseTranslations,
+  baseTranslate,
+  togglePassword,
+  getUserCtx
+} from './auth-utils.js';
+
+let selectedLocale;
+let translations;
+
+const PASSWORD_INPUT_ID = 'password';
+const CONFIRM_PASSWORD_INPUT_ID = 'confirm-password';
+
+const translate = () => {
+  baseTranslate(selectedLocale, translations);
+};
+
+const displayPasswordValidationError = (serverResponse) => {
+  const { error, params } = JSON.parse(serverResponse);
+  setState(error);
+
+  const passwordError = document.querySelector('.error.password-short');
+  if (params?.minimum && passwordError) {
+    passwordError.setAttribute('translate-values', JSON.stringify(params));
+    baseTranslate(selectedLocale, translations);
+  }
+};
+
+const submit = function(e) {
+  e.preventDefault();
+  if (document.getElementById('form')?.className === 'loading') {
+    // debounce double clicks
+    return;
+  }
+  setState('loading');
+  const url = document.getElementById('form')?.action;
+  const password = document.getElementById(PASSWORD_INPUT_ID)?.value;
+  const confirmPassword = document.getElementById(CONFIRM_PASSWORD_INPUT_ID)?.value;
+  const userCtx = getUserCtx();
+
+  const payload = JSON.stringify({
+    user: userCtx.name,
+    password: password,
+    confirmPassword: confirmPassword,
```

Comment:
There's no point submitting both the `password` and the `confirmPassword`. Just do a quick `===` check on the client side before submitting and just send through one copy. This is better UX because it can fail without the round trip to the server.

Ideally we'd also check password strength on the client side but if it's too hard to share the password strength library on the client side then it's fine to leave this for another day.